src/GitHubCreateMergePRs/Program.cs

```diff
@@ -157,4 +156,66 @@ private static async Task RunAsync(XDocument config, bool isAutomatedRun, bool i
             }
         }
     }
+
+    /// <summary>
+    /// Checks the merge element for a frequency attribute. Then determines whether the current run
+    /// matches the frequency criteria. Valid frequency values are 'daily' and 'weekly'.
+    /// </summary>
+    private static bool ShouldRunMerge(XElement merge, bool isAutomatedRun)
+    {
+        // We always run when merges are started manually
+        if (!isAutomatedRun)
+        {
+            return true;
+        }
+
+        var frequency = merge.Attribute("frequency")?.Value.ToLower();
+
+        // We always run when a frequency isn't specified
+        if (string.IsNullOrEmpty(frequency))
+        {
+            return true;
+        }
+
+        // Throw when an unexpected frequency value is specified
+        if (frequency != "daily" && frequency != "weekly")
+        {
+            throw new Exception($"Unexpected merge frequency specified: '{frequency}'. Valid values are 'daily' and 'weekly'.");
+        }
+
+        // Since this is run on AzDO as an automated cron pipeline, times are in UTC.
+        // See https://docs.microsoft.com/en-us/azure/devops/pipelines/build/triggers?view=azure-devops&tabs=yaml#scheduled-triggers
+        var runDateTime = DateTime.UtcNow;
+
+        // Since cron should schedule this to run every 3 hours starting at 12am,
+        // we expect to be run within a 10 minute window of this time.
+
+        // Adjust the time 5 minutes into the future in case the pipeline machine
+        // and scheduler machine have mismatched clocks.
+        var adjustedRunDateTime = runDateTime.AddMinutes(5);
+        var adjustedRunDate = adjustedRunDateTime.Date;
+
+        // Because of the adjusted run time is being used we treat a run as valid
+        // if it begins in the last 5 minutes of the previous day through the first
+        // 5 minutes of the current day.
+        var tenMinutes = new TimeSpan(hours: 0, minutes: 10, seconds: 0);
+        var isStartOfDay = adjustedRunDateTime - adjustedRunDate < tenMinutes;
+
+        // Daily and Weekly runs only happen at the start of the day
+        if (!isStartOfDay)
+        {
+            return false;
+        }
+
+        var isSunday = adjustedRunDate.DayOfWeek != DayOfWeek.Sunday;
```

Comment:
Please do this inside the body of the `if (frequency == "weekly")`.

Also, it looks like this condition should maybe be negated.